app/activeplan/homepage/get.controller.js

```diff
@@ -1,26 +1,51 @@
 const { logger } = require('../../../common/logging/logger')
 const { getSentencePlan, getSentencePlanMeetings } = require('../../../common/data/sentencePlanningApi')
 const { getCommentText } = require('../../../common/utils/getCommentText')
+const { groupBy } = require('../../../common/utils/util')
 
 const getHomepage = async (req, res) => {
   const {
     session: { planStarted = false, 'x-auth-token': token },
     params: { planId, id },
   } = req
 
-  let sentencePlan
   let meetings
+  let comments
+  let objectives
 
   delete req.session.planStarted
 
   // get the sentence plan
   try {
-    sentencePlan = await getSentencePlan(planId, token)
+    ;({ comments, objectives } = await getSentencePlan(planId, token))
   } catch (error) {
     logger.error(`Could not retrieve active sentence plan ${planId} for offender ${id}, error: ${error}`)
     return res.render('app/error', { error })
   }
 
+  // determine the status of each objective
+  objectives = objectives.map(objective => {
+    const currentObjective = objective
+
+    // objectives default to active if not caught with the rules below
+    currentObjective.type = 'active'
+
+    if (currentObjective.actions.every(action => ['NOT_STARTED'].includes(action.status))) {
+      currentObjective.type = 'future'
+    } else if (
+      currentObjective.actions.every(action =>
+        ['COMPLETED', 'PARTIALLY_COMPLETED', 'ABANDONED'].includes(action.status)
+      )
```

Comment:
Not for this, but I think we need to create a some status constants and utils 👍 
Its referenced in multiple places across the app.

Reply:
agreed